pkg/systemd/services/service-details.jsx

```diff
@@ -278,35 +293,89 @@ export class ServiceDetails extends React.Component {
         }
     }
 
-    unitAction(method, extra_args) {
+    unitAction(method, extra_args, catchExc = true) {
         if (extra_args === undefined)
             extra_args = ["fail"];
         this.setState({ waitsAction: true });
-        systemd_client[this.props.owner].call(SD_OBJ, SD_MANAGER, method, [this.props.unit.Names[0]].concat(extra_args))
-                .catch(error => {
-                    this.show_error(error.toString());
-                    this.setState({ waitsAction: false });
-                });
+        const promise = systemd_client[this.props.owner].call(SD_OBJ, SD_MANAGER, method, [this.props.unit.Names[0]].concat(extra_args));
+        if (catchExc) {
+            return promise.catch(error => {
+                this.show_error(error.toString());
+                this.setState({ waitsAction: false });
+            });
+        } else {
+            return promise;
+        }
     }
 
-    unitFileAction(method, force) {
+    unitFileAction(method, force, catchExc = true) {
         this.setState({ waitsFileAction: true });
         const args = [[this.props.unit.Names[0]], false];
         if (force !== undefined)
             args.push(force == "true");
-        systemd_client[this.props.owner].call(SD_OBJ, SD_MANAGER, method, args)
+        const promise = systemd_client[this.props.owner].call(SD_OBJ, SD_MANAGER, method, args)
                 .then(([results]) => {
                     if (results.length == 2 && !results[0])
                         this.show_note(_("This unit is not designed to be enabled explicitly."));
                     /* Executing daemon reload after file operations is necessary -
                      * see https://github.com/systemd/systemd/blob/main/src/systemctl/systemctl.c [enable_unit function]
                      */
                     systemd_client[this.props.owner].call(SD_OBJ, SD_MANAGER, "Reload", null);
-                })
-                .catch(error => {
-                    this.show_error(error.toString());
-                    this.setState({ waitsFileAction: false });
                 });
+        if (catchExc) {
+            return promise.catch(error => {
+                this.show_error(error.toString());
+                this.setState({ waitsFileAction: false });
+            });
+        } else {
+            return promise;
+        }
+    }
+
+    deleteAction() {
+        this.getUnitPaths().then(unitPaths => {
+            this.setState({ showDeleteDialog: true, unitPaths });
+        });
+    }
+
+    async getUnitPaths() {
+        const paths = [this.props.unit.FragmentPath];
+
+        await Promise.all(this.props.unit.Triggers.map(async trigger => {
+            // Getting dbus properties from a non-loaded unit is not possible so resort to systemctl show
```

Comment:
FTR, that's not really true -- systemd has a [LoadUnit()](https://www.freedesktop.org/wiki/Software/systemd/dbus/) for exactly that (after all, systemctl also just uses the D-Bus protocol).

Reply:
.. but, calling systemctl here is acceptable here. It's much more expensive, but it's not going to be that many calls.